src/NationalInstruments.Analyzers/Style/ChainOfMethodsWithLambdasAnalyzer.cs

```diff
@@ -0,0 +1,110 @@
+            Resources.CategoryNI,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecutionIf(IsRunningInProduction);
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSyntaxNodeAction(
+                AnalyzeSyntaxNode,
+                SyntaxKind.ExpressionStatement,
+                SyntaxKind.EqualsValueClause,
+                SyntaxKind.Argument,
+                SyntaxKind.ArrowExpressionClause);
+        }
+
+        private static void AnalyzeSyntaxNode(SyntaxNodeAnalysisContext context)
+        {
+            var parentSyntaxNode = context.Node;
+
+            // Find the invocation expression i.e
+            // a method/delegate call or a property access
+            // or a chain of them
+            var invocationExpressionSyntax = parentSyntaxNode
+                .DescendantNodes()
+                .OfType<InvocationExpressionSyntax>()
+                .FirstOrDefault();
+
+            if (invocationExpressionSyntax is null)
+            {
+                // This does not contain any invocations, bail out
+                return;
+            }
+
+            bool IsLambdaExpression(SyntaxNode syntaxNode) => syntaxNode.IsKind(SyntaxKind.SimpleLambdaExpression);
+
+            bool IsNotArgumentSyntax(SyntaxNode syntaxNode) => !syntaxNode.IsKind(SyntaxKind.Argument);
+
+            // Find all non-nested arguments which contain a lambda expression
+            var nonNestedArgumentsWithLambdas = invocationExpressionSyntax
+                .DescendantNodes(IsNotArgumentSyntax)
+                .OfType<ArgumentListSyntax>()
+                .Where(argument => argument.DescendantNodes().Any(IsLambdaExpression));
+
+            // There is none or only a single call
+            if (nonNestedArgumentsWithLambdas.Count() < 2)
+            {
+                // No more refactoring required
+                return;
+            }
+
+            // The first argument can stay on the same line
+            var successiveArgumentsWithLambdas = nonNestedArgumentsWithLambdas.Skip(1);
+
+            foreach (var argument in successiveArgumentsWithLambdas)
+            {
+                // Get the parent invocation expression
+                var parentInvocation = argument.Parent;
+
+                // Find the method/delegate call
+                var memberAccessExpression = parentInvocation?.ChildNodes().OfType<MemberAccessExpressionSyntax>().FirstOrDefault();
+
+                if (memberAccessExpression is null)
+                {
+                    // unknown syntax, don't know what to do
```

Comment:
We could throw an exception here which will bubble up as AD0001 so clients can report that we haven't considered all cases. The alternative would be them seeing chained methods with lambdas on the same line not being reported and tell us, but that seems unlikely

Reply:
Done.